aten/src/ATen/native/cuda/Indexing.cu

```diff
@@ -567,5 +568,258 @@ Tensor& index_add_cuda_(Tensor & self, int64_t dim, const Tensor & index, const
 #undef LARGE_INDEX
 }
 
-} //at
-} //native
+namespace {
+// We prefer this kernel to avoid reloading index points if the number
+// of indices is a small number.
+// This kernel in fact works for all choices of problem size, but if
+// the number of indices chosen is large, then the
+// indexSelectLargeIndex kernel is a better choice to increase
+// parallelism.
+// TODO: remove
```

Comment:
Can you elaborate on this TODO?

Reply:
Sorry, it was a note-to-self that's no longer relevant. I forgot to remove it.